### PR TITLE
Issue #11782: releasenotes script generated empty commit

### LINKED ISSUE #11782: releasenotes script generated empty commit
https://github.com/checkstyle/checkstyle/commit/b8d8466aa4db10128990a64384a88450d3b2e498

```
$ git show
commit b8d8466aa4db10128990a64384a88450d3b2e498
Author: Roman Ivanov <ivanov-jr@mail.ru>
Date:   Mon Jun 27 06:43:04 2022 -0700

    doc: release notes 10.3.1-SNAPSHOT

```

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves #11782 

Child issue: https://github.com/checkstyle/contribution/pull/618

### CODE REVIEW

File: .ci/releasenotes-gen-xdoc-push.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.
      -githubAuthToken "$READ_ONLY_TOKEN" \
      -generateXdoc \
      -xdocTemplate $BUILDER_RESOURCE_DIR/templates/xdoc_freemarker.template \
-     -publishXdoc -publishXdocWithPush
+     -publishXdoc
```
Comment (romani) [inline .ci/releasenotes-gen-xdoc-push.sh:54]: Ok, but please do commit by shell commands

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/releasenotes-gen-xdoc-push.sh (hunk 1 above)
Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:54]: This script must all work.

I presume `publishXdoc** - (optional) Make commit in local checkstyle repo with releasenotes.` is place where there i problem, so better to do this by shell commands, that very robust.

**Comment by Rahulkhinchi03** [comment]
```
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.223 s
[INFO] Finished at: 2022-07-06T19:49:13+05:30
[INFO] ------------------------------------------------------------------------
[auto e84056dbb] config: update version to 10.4
 1 files changed, 2 insertions(+), 3 deletions(-)
To github.com:Rahulkhinchi03/checkstyle.git

```

**Comment by romani** [comment]
When you commit, you create commit on your local, GitHub completely unaware of new commit. As you do push, git sync commits to GitHub repository, so GitHub can show something in web page.


Please remove extra version bump, squash commits and prepare for review.

File: .ci/releasenotes-gen-xdoc-push.sh [hunk 2]
```diff
@@ -51,7 +51,17 @@ java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.
...
+CS_RELEASE_VERSION=$(mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
+              -Dexec.args='${project.version}' \
+              --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec | sed 's/-SNAPSHOT//')
+  
+LATEST_RELEASE_TAG=$(curl -s https://api.github.com/repos/checkstyle/checkstyle/releases/latest \
+                       | jq ".tag_name")
+
+mvn -e --no-transfer-progress versions:set -DgroupId=com.puppycrawl.tools -DartifactId=checkstyle \
+  -DoldVersion=10.4-SNAPSHOT -DnewVersion="$CS_RELEASE_VERSION"
+
```
Comment (romani) [inline .ci/releasenotes-gen-xdoc-push.sh:66]: Remove this bump,.
Version bump is done by release.sh during mvn release:perform.

All you need in this PR is just update xdoc and commit and push it.


Please share prove of how it works in your repo.

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/releasenotes-gen-xdoc-push.sh (hunk 2 above)
Comment (Rahulkhinchi03) [reply .ci/releasenotes-gen-xdoc-push.sh:66]: @romani I am a little confused, then what should the commit contain ( if not the change of project version)?

Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:66]: It should contain xdoc update - release notes

Comment (romani) [reply .ci/releasenotes-gen-xdoc-push.sh:66]: Example https://github.com/checkstyle/checkstyle/commit/47e6e280027940ba741d552ab9147fba7b4152a7

File: .ci/releasenotes-gen-xdoc-push.sh [hunk 3]
```diff
@@ -50,8 +50,12 @@ java -jar contribution/releasenotes-builder/target/releasenotes-builder-1.0-all.
      -releaseNumber "$CS_RELEASE_VERSION" \
      -githubAuthToken "$READ_ONLY_TOKEN" \
      -generateXdoc \
-     -xdocTemplate $BUILDER_RESOURCE_DIR/templates/xdoc_freemarker.template \
-     -publishXdoc -publishXdocWithPush
+     -xdocTemplate $BUILDER_RESOURCE_DIR/templates/xdoc_freemarker.template
+
+cd ../
+sed -i '12r .ci-temp/xdoc.xml' src/xdocs/releasenotes.xml
+git add . && git commit -m "config: update version to $CS_RELEASE_VERSION"
```
Comment (romani) [inline .ci/releasenotes-gen-xdoc-push.sh:57]: Commit message: "doc: release notes for {version}"

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .ci/releasenotes-gen-xdoc-push.sh (hunk 3 above)
Comment (Rahulkhinchi03) [reply .ci/releasenotes-gen-xdoc-push.sh]: Changes Done

**Comment by romani** [comment]
Please show how bit works in your repo .

To prove that it works.

**Comment by Rahulkhinchi03** [comment]
> Please show how bit works in your repo .
> 
> To prove that it works.

The oroof link of commit generated: https://github.com/checkstyle/checkstyle/commit/f20f74953caa5d6b88171350839aa410606fd493

**Pull request merged**